drivers/volume/portworx/portworx.go

```diff
@@ -949,53 +952,201 @@ func (p *portworx) SnapshotDelete(snapDataSrc *crdv1.VolumeSnapshotDataSource, _
 	}
 }
 
-// VolumeSnapshotRestore does in-place restore of snapshot to it's parent volume
-func (p *portworx) VolumeSnapshotRestore(snapRestore *stork_crd.VolumeSnapshotRestore, restoreVolumes map[string]string) error {
+// StartVolumeSnapshotRestore will prepare volume for restore
+func (p *portworx) StartVolumeSnapshotRestore(snapRestore *stork_crd.VolumeSnapshotRestore) error {
+	volumeInfos := make([]*stork_crd.RestoreVolumeInfo, 0)
+	volRestores, snapType, credID, err := processRestoreVolumes(snapRestore.Status.RestoreVolumes)
+	if err != nil {
+		logrus.Errorf("Invalid snapshot data %v", err)
+		return err
+	}
+
+	switch snapType {
+	case "", string(crdv1.PortworxSnapshotTypeLocal):
+		snapRestore.Status.Volumes = volumeInfos
+		snapRestore.Status.RestoreType = string(crdv1.PortworxSnapshotTypeLocal)
+		return nil
+	case string(crdv1.PortworxSnapshotTypeCloud):
+		// Get volume client from user context
+		volDriver, err := p.getUserVolDriver(snapRestore.Annotations)
+		if err != nil {
+			return err
+		}
+		for volID, snapID := range volRestores {
+			// Get Volume Info
+			vols, err := volDriver.Inspect([]string{volID, restoreNamePrefix + volID})
```

Comment:
Why is `restoreNamePrefix + volID` being used in the Inspect()?

Reply:
we are downloading restore object to volume ```restoreNamePrefix + volID```